core/src/main/java/greencity/validator/SocialNetworkLinksValidator.java

```diff
@@ -0,0 +1,36 @@
+package greencity.validator;
+
+import greencity.annotations.ValidSocialNetworkLinks;
+import greencity.constant.ErrorMessage;
+import greencity.constant.ValidationConstants;
+import greencity.exception.exceptions.BadSocialNetworkLinksException;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+
+import static greencity.constant.ErrorMessage.USER_CANNOT_ADD_MORE_THAN_5_SOCIAL_NETWORK_LINKS;
+
+public class SocialNetworkLinksValidator implements ConstraintValidator<ValidSocialNetworkLinks, List<String>> {
+    @Override
+    public void initialize(ValidSocialNetworkLinks constraint) {
+    }
+
+    @Override
+    public boolean isValid(List<String> links, ConstraintValidatorContext context) {
+        if (links.size() > ValidationConstants.MAX_AMOUNT_OF_SOCIAL_NETWORK_LINKS) {
+            throw new BadSocialNetworkLinksException(USER_CANNOT_ADD_MORE_THAN_5_SOCIAL_NETWORK_LINKS);
+        }
+        if (!areDistinct(links)) {
+            throw new BadSocialNetworkLinksException(ErrorMessage.USER_CANNOT_ADD_SAME_SOCIAL_NETWORK_LINKS);
+        }
+        links.forEach(UrlValidator::isUrlValid);
+        return true;
+    }
+
+    private boolean areDistinct(List<String> list) {
+        Set<String> hashSet = new HashSet<>(list);
```

Comment:
I'm not sure it is a good idea to check if links are distinct. You can convert behind the scene the `List` to a `Set` so they are distinct and do not borrow user to search which link has a duplicate. Then check `links.size() > ValidationConstants.MAX_AMOUNT_OF_SOCIAL_NETWORK_LINKS`. Please review requirements once again.

Reply:
![image](https://user-images.githubusercontent.com/29165636/93058391-01819780-f678-11ea-8121-10b40d431666.png)

Reply:
What I can do about this?